src/client/pkg/grpcutil/server.go

```diff
@@ -7,67 +7,66 @@ import (
 	"net"
 	"time"
 
-	"github.com/pachyderm/pachyderm/src/client/version"
-	"github.com/pachyderm/pachyderm/src/client/version/versionpb"
-
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/keepalive"
 )
 
 var (
 	// ErrMustSpecifyRegisterFunc is used when a register func is nil.
 	ErrMustSpecifyRegisterFunc = errors.New("must specify registerFunc")
-)
 
-// ServeOptions represent optional fields for serving.
-type ServeOptions struct {
-	Version    *versionpb.Version
-	MaxMsgSize int
-	Cancel     chan struct{}
-}
+	// ErrMustSpecifyPort is used when a port is 0
+	ErrMustSpecifyPort = errors.New("must specify port on which to serve")
+)
 
-// ServeEnv are environment variables for serving.
-type ServeEnv struct {
-	// Default is 7070.
-	GRPCPort uint16 `env:"GRPC_PORT,default=7070"`
+// ServerSpec represent optional fields for serving.
+type ServerSpec struct {
+	Port         uint16
+	MaxMsgSize   int
+	Cancel       chan struct{}
+	RegisterFunc func(*grpc.Server) error
 }
```

Comment:
`ServerOptions` seems like a better name for this than `ServerSpec` to me, was there a reason you changed it?